galite-core/src/main/kotlin/org/kopi/galite/visual/VItemTree.kt

```diff
@@ -15,6 +15,285 @@
+      }
+      CMD_REMOVE -> if (ask(Message.getMessage("confirm_delete"), false)) {
+        currentDisplay.removeSelectedItem()
+        refresh()
+      }
+      CMD_EDIT -> {
+        currentDisplay.editSelectedItem()
+        refresh()
+      }
+      CMD_FOLD -> currentDisplay.getTree().collapseRow(currentDisplay.getTree().getSelectionRow())
+      CMD_UNFOLD -> currentDisplay.getTree().expandRow(currentDisplay.getTree().getSelectionRow())
+      CMD_DEFAULT -> currentDisplay.setDefaultItem()
+      CMD_LOCALISE -> {
+        currentDisplay.localiseSelectedItem()
+        refresh()
+      }
+      CMD_SAVE -> if (isChanged && itemTreeManager != null) {
+        setWaitInfo("")
+        itemTreeManager.save()
+        unsetWaitInfo()
+        isChanged = false
+        currentDisplay.setTree()
+      }
+      else -> super.executeVoidTrigger(key)
+    }
+  }
+
+  /**
+   * Refresh the item Tree view
+   */
+  fun refresh() {
+    isChanged = true
+    getDisplay().setTree()
+  }
+
+  /**
+   * Builds the item tree.
+   */
+  private fun createTree() {
+    rootItem = RootItem(-1, rootName)
+    rootItem!!.createTree(items)
+    root = rootItem!!.getRoot()
+  }
+
+  /**
+   * Return items array
+   */
+  fun getItems(): Array<Item> = getDisplay().getTree().getItems()
+
+  /**
+   * Return the root item
+   */
+  fun getRootItem(): Item = getDisplay().getTree().getRootItem()
+
+  /**
+   * Init value of max ID
+   */
+  private fun initMaxId() {
+    maxId = -1
+    for (i in items.indices) {
+      if (maxId < items[i].getId()) {
+        maxId = items[i].getId()
+      }
+    }
+  }
+
+  // --------------------------------------------------------------------
+  // ACCESSORS
+  // --------------------------------------------------------------------
+  override fun getType(): Int = Constants.MDL_ITEM_TREE
+
+  override fun getDisplay(): UItemTree = super.getDisplay() as UItemTree
+
+  fun isNoEdit(): Boolean = selectionType == NO_EDIT
+
+  fun isSingleSelection(): Boolean = selectionType == SINGLE_SELECTION
+
+  fun isMultiSelection(): Boolean = selectionType == MULTI_SELECTION
+
+  fun isMultiSelectionDefaultValue(): Boolean = selectionType == MULTI_SELECTION_DEFAULT_VALUE
+
+
+  val nextId: Int
+    get() {
+      maxId++
+      return maxId
+    }
+
+  // --------------------------------------------------------------------
+  // DATA MEMBERS
+  // --------------------------------------------------------------------
+  /**
+   * Returns the root Item of this tree.
+   * @return The root node of this tree.
+   */
+  var root: TreeNode? = null
+    private set
+  private var rootItem: RootItem? = null
+  private val actors: Array<VActor>? = null
+  private var maxId = 0
+  private val rootName: String = rootName ?: "Items"
+
+  /**
+   * Set model changed
+   */
+  var isChanged: Boolean = false
+
+  /**
+   * Return the removed items list
+   */
+  val removedItems: List<Item> = ArrayList()
+
+  init {
+    rootName?.let { setTitle(it) }
```

Comment:
if you eliminated "?" , change it to setTitle(rootName)

Reply:
rootName is nullable